### ISSUE TITLE
bug: statistic are sent from NMS entity even if its stat collector is not started

### ISSUE THREAD

**Issue by mathieucarbou** [Terracotta-OSS/terracotta-platform#395]
In the case where several nms entities are created, each nms entity will receive all stats generated from every stat colletors, even if they are not started.

I.e. if you connect 2 NMS clients, and start collecting each 1 sec in one and each 5 sec in the other one, then both will receive all the collected stats at both 1 sec and 5 sec interval.

Also, if 2 NMS clients are connected and only 1 start collecting stats, they are all receiving the collected stats from the started collector, even if one collector is not started.

@anthonydahanne : FYI

**Comment by anthonydahanne** [comment]
"Also, when using passives, if 2 NMS clients are connected and only 1 start collecting stats, then all are receiving the stats and they are duplicated." --> Not sure to understand ... clients connected to passives ? can you please clarify this last statement ? thanks

**Comment by mathieucarbou** [comment]
@anthonydahanne : I updated to make it clearer, it does not depend on passives sorry ;-)

**Comment by mathieucarbou** [comment]
I've found the issue, and found a patch also, but by testing the patch, I found another bug:

The `serverBecomeActive` event that we receive now from the platform is received AFTER that all the active entities have been created.

In the case of a failover, this is problematic because all the active entities are created BEFORE the server sent its `serverDidBecomeActive` events to M&M, which is the place where we do some cleanup stuff. 

This cleanup is required because when promoting an entity, platform does not call destroy on existing passing entities but instead just create new active entities.

**Comment by mathieucarbou** [comment]
Finally there were 2 bugs: 

1st bug: all stat collectors where sending to all NMS entities, not just the one that created them.

when fixing this bug, I found that the stats were not received after a failover

2nd bug: because after a failover, the newly created registry for the active entity was destroyed just after. BUT we saw nothing because the one of the passive entity was just still running on the background.
